precommit/src/main/shell/test-patch.d/maven.sh

```diff
@@ -14,7 +14,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-declare -a MAVEN_ARGS
+# Accept a value of MAVEN_ARGS from the calling context
+# shellcheck disable=SC2206
```

Comment:
Shellcheck tells me that I need to quote everything inside the parentheses, but I want `MAVEN_ARGS` to be interpreted as an array. If I quote it, it doesn't do the splitting I'm after.

```sh
$ x='hello there'
$ unset y
$ declare -a FOO=(${x:-})
$ set | grep FOO
FOO=([0]="hello" [1]="there")
_=FOO
$ declare -a FOO=("${x:-}")
$ set | grep FOO
FOO=([0]="hello there")
_=FOO
```

Reply:
Why are we trying to get an initial MAVEN_ARGS from the environment? the use of personality in e.g. HBASE-23779 should be adding to an array we declare here, not creating a new environment variable that we need to parse.

Reply:
This is what I pushed last night into hbase-personality... it don't work either.

-  export MAVEN_ARGS="-T0.5C ${MAVEN_ARGS}"
+  MAVEN_ARGS=("-T0.5C" "${MAVEN_ARGS[@]}")

Reply:
were these two different attempts? I'd expect the first to fail and the second to succeed.

Reply:
Stack's addendum was included in https://builds.apache.org/job/HBase%20Nightly/job/master/1622/

I don't see his tweaked MAVEN_ARGS in any of the build invocation logs, for example, https://builds.apache.org/job/HBase%20Nightly/job/master/1622/artifact/output-jdk8-hadoop2/patch-compile-root.txt

Reply:
> Why are we trying to get an initial MAVEN_ARGS from the environment? the use of personality in e.g. HBASE-23779 should be adding to an array we declare here, not creating a new environment variable that we need to parse.

@busbey you saying we have the order of operations wrong? Personality appends to the base plugin?

Reply:
Issue HBASE-23779 has the jobs I ran after pushing 'fixes'.  hopefully: https://builds.apache.org/view/H-L/view/HBase/job/HBase%20Nightly/job/branch-2/2452/ and then 2454 is the one that ran w/ second attempt.

Reply:
Ugh. Hbase nightly saves the entire Yetus install in build artifacts but not the actual personality file used.

Lemme see what this looks like locally.

Reply:
> @busbey you saying we have the order of operations wrong? Personality appends to the base plugin?

Yes. The docs say `personality_globals` is done after importing plugins.

Reply:
Thanks @busbey (and @ndimiduk )